app/src/main/java/org/covidwatch/android/CovidWatchApplication.kt

```diff
@@ -28,20 +37,34 @@ class CovidWatchApplication : Application() {
     }
 
     private fun schedulePeriodicPublicContactEventsRefresh() {
-        val constraints = Constraints.Builder()
-            .setRequiresCharging(false)
-            .setRequiredNetworkType(NetworkType.CONNECTED)
-            .build()
-
         val downloadRequest =
             PeriodicWorkRequestBuilder<ContactEventsDownloadWorker>(1, TimeUnit.HOURS)
-                .setConstraints(constraints)
+                .setConstraints(getRefreshConstraints())
                 .build()
 
         WorkManager.getInstance(this).enqueueUniquePeriodicWork(
-            ContactEventsDownloadWorker.WORKER_NAME,
+            "ContactEventsDownloadWorkerPeriodicRefresh",
             ExistingPeriodicWorkPolicy.REPLACE,
             downloadRequest
         )
     }
+
+    fun executePublicContactEventsRefresh() {
+        val downloadRequest = OneTimeWorkRequestBuilder<ContactEventsDownloadWorker>()
+            .setConstraints(getRefreshConstraints())
+            .build()
+
+        WorkManager.getInstance(this).enqueueUniqueWork(
+            "ContactEventsDownloadWorkerOneTimeRefresh",
```

Comment:
When you move this code to separate classes it's better to have this string as a constant. Then, if would need to cancel job we just reuse the constant.